### DISCUSSION TITLE
editable_select trigger on blur BUG or ?

### DISCUSSION THREAD

**Discussion by AdamovicBoban** [Laravel-Backpack/community-forum#762] [Q&A (Help)] [answered]
I turned off   
       `
        ->save_on_focusout(false)
        ->save_on_change(false)
        ->auto_update_row(false)
`

but it is executed onblur.
Can someone check and confirm if it's a bug?

https://github.com/Laravel-Backpack/community-forum/assets/68257663/c0badc11-5968-4e88-beb1-17a7b0ee29d5

**Comment by AdamovicBoban** [comment]
Yes, this is a BUG. When adding a wrapper to the editable_select, the update route is lost. Please have someone check and confirm. 
I also noticed that when adding a custom attribute via the wrapper, it doesn't add it to the select; instead, it creates an 'a' tag and inserts the select into it.

![image](https://github.com/Laravel-Backpack/community-forum/assets/68257663/7948ca67-e271-47b1-95df-27e32765aa9d)

**Comment by karandatwani92** [comment]
Hey @pxpm,

I tested this with a simple wrapper. YES, it BREAKS the column. URL becomes undefined because of new element in between. I tried the `span` element. Same ERROR!

```php
CRUD::column('select_from_array')
            ->label('Select')
            ->type('editable_select')
            ->options(['one' => 'One', 'two' => 'Two', 'three' => 'Three'])
            ->wrapper(['class' => 'bg-primary',]);
```

And, I tried to transfer the issue to [Laravel-Backpack/editable-columns](https://github.com/Laravel-Backpack/editable-columns).
But it was not coming on my list.

**Reply by AdamovicBoban** [reply]
Yes, a real issue when adding a wrapper. Even if you turn everything to false  `->save_on_focusout(false)->save_on_change(false)->auto_update_row(false)` and remove the wrapper, it still performs an update onBlur.

The solution for the wrapper is in `vendor/backpack/editable-columns/resources/views/miscellaneous/minor_update_client_logic.blade.php` 
in `function selectEventsHandler()` add at the beginning: `let container = input.closest('[data-column-editable]');` and everywhere in this function, replace `input.parentElement` with `container`.

**Comment by pxpm** [comment, marked as answer]
Hey @AdamovicBoban, thanks for the report. 

I've just tagged editable columns 3.0.3 with the fix for this issue. 

Regarding the `save_on_focus(false)` for the select columns, that wouldn't have any effect, because the select column can only be saved in two scenarios: `change` or `blur/focus out`. 

You can disable the `save_on_change` but it will always save on blur as a fallback. Contrary to other column types that can be saved with `Enter` key if you disable the change and blur saving. `

If you want to disable the save on change and save on blur in this column I think you probably don't want an editable column in that scenario. 

Thanks again for the report @AdamovicBoban and @karandatwani92 for acknowledging the issue 🙏 

Our private repo should take a couple of hours to pick the change, but then you are able to get the fix with a `composer update`. 

Cheers
